vortex-scan/src/range_scan.rs

```diff
@@ -106,8 +107,16 @@ impl RangeScan {
                 let mask = result.into_bool()?.boolean_buffer();
                 let mask = self.mask.to_boolean_buffer()?.bitand(&mask);
                 // Then move onto the projection
-                self.state =
-                    State::Project((FilterMask::from(mask), self.scan.projection().clone()))
+                if mask.is_empty() {
+                    // If the mask is empty, then we're done.
+                    let len = mask.len();
```

Comment:
Should we have a dedicated function for `null_array(len)`?

Reply:
Why do you have to return an array at all in this case?

Reply:
basically seems like it would be better if `Ready` was `Option<ArrayData>`

Reply:
At some point I need to return an array to the user, is there a way to create a null constant array of dtype.

Reply:
Why do you want to return a null array? If everything has been filtered out we can return array of length 0?

Reply:
I think if your mask is empty then the result is a 0 len array so you can return `Canonical::empty`